.github/workflows/publish.yml

```
@@ -0,0 +1,86 @@
+name: Publish package to npmjs
+
+# This workflow runs when code is pushed to `main` (i.e: when a pull request is merged)
+on:
+    push:
+        branches: [main]
+
+# Ensure that only one instance of this workflow executes at a time.
+# If multiple PRs are merged in quick succession, there will only ever be one publish workflow running and one pending.
+concurrency: ${{ github.workflow }}
+
+jobs:
+    version:
+        runs-on: ubuntu-latest
+
+        # OSBotify will update the version on `main`, so this check is important to prevent an infinite loop
+        if: ${{ github.actor != 'OSBotify' }}
+
+        steps:
+            - uses: actions/checkout@v3
+              with:
+                ref: main
```

Gotta pass OSBotify token to `actions/checkout` or else pushing to protected branch won't work. It's an org secret so this should just work:

```suggestion
                ref: main
                # The OS_BOTIFY_COMMIT_TOKEN is a personal access token tied to osbotify, which allows him to push to protected branches
                token: ${{ secrets.OS_BOTIFY_COMMIT_TOKEN }}
```

---

Oh I have an outdated version of this workflow. I'll update it to match the one used in Onyx